Source/Laser/LaserProfilesImpl/LaserProfileFromFile.cpp

```diff
@@ -0,0 +1,570 @@
+/* Copyright 2019-2020 Luca Fedeli
+ * Ilian Kara-Mostefa
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#include "Laser/LaserProfiles.H"
+
+#include "Utils/Algorithms/LinearInterpolation.H"
+#include "Utils/Algorithms/UpperBound.H"
+#include "Utils/Parser/ParserUtils.H"
+#include "Utils/TextMsg.H"
+#include "Utils/WarpX_Complex.H"
+#include "Utils/WarpXConst.H"
+
+#include <ablastr/warn_manager/WarnManager.H>
+
+#include <AMReX.H>
+#include <AMReX_Algorithm.H>
+#include <AMReX_Config.H>
+#include <AMReX_Extension.H>
+#include <AMReX_GpuContainers.H>
+#include <AMReX_GpuDevice.H>
+#include <AMReX_GpuLaunch.H>
+#include <AMReX_GpuQualifiers.H>
+#include <AMReX_PODVector.H>
+#include <AMReX_ParallelDescriptor.H>
+#include <AMReX_ParmParse.H>
+#include <AMReX_Print.H>
+#include <AMReX_REAL.H>
+#include <AMReX_Vector.H>
+
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+#include <fstream>
+#include <iterator>
+#include <limits>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+// example: data handling & print
+#include <vector>   // std::vector
+#include <iostream> // std::cout
+#include <memory>   // std::shared_ptr
```

Comment:
I would suggest removing these comments:

```suggestion
#include <vector>
#include <iostream>
#include <memory>
```